streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -205,10 +205,12 @@ private[streaming] class FileBasedWriteAheadLog(
 
   /** Stop the manager, close any open log writer */
   def close(): Unit = synchronized {
-    if (currentLogWriter != null) {
-      currentLogWriter.close()
+    if (!executionContext.isShutdown) {
```

Comment:
Does this need to be thread-safe? it seems like the impl just above tries to be. I don't know if it's needed.